src/emulator/modules/SceCtrl/SceCtrl.cpp

```diff
@@ -146,35 +146,58 @@ static uint8_t float_to_byte(float f) {
     return static_cast<uint8_t>(clamped * 255);
 }
 
+static int reserve_port(CtrlState &state) {
+    int res = -1;
+    for (int i = 0; i < 4; i++) {
+        if (state.free_ports[i]) {
+            state.free_ports[i] = false;
+            res = i + 1;
+            break;
+        }
+    }
+    return res;
+}
+
 static void remove_disconnected_controllers(CtrlState &state) {
-    for (GameControllerList::iterator controller = state.controllers.begin(); controller != state.controllers.end();) {
-        if (SDL_GameControllerGetAttached(controller->second.get())) {
+    for (ControllerList::iterator controller = state.controllers.begin(); controller != state.controllers.end();) {
```

Comment:
```suggestion
    for (auto controller : controllers) {
```
and change usage accordingly if needed